File: TelerikUIExamplesInCSharp/Examples/Alert/AlertSettings.cs
using System;

using Foundation;
using UIKit;
using CoreGraphics;
using TelerikUI;

namespace Examples
{
	[Register("AlertSettings")]
	public class AlertSettings : XamarinExampleViewController
	{
		TKDataFormEntityDataSourceHelper dataSource;
		TKDataForm dataForm;
		Settings settings;
			
		public override void ViewDidLoad ()
		{
			this.AddOption ("Show Alert", ShowAlert);

			base.ViewDidLoad ();

			this.AutomaticallyAdjustsScrollViewInsets = false;

			settings = new Settings ();
			this.dataSource = new TKDataFormEntityDataSourceHelper (settings);

			dataSource["Title"].Index = 0;
			dataSource["Message"].Index = 1;
			dataSource["AllowParallax"].Index = 2;
			dataSource["BackgroundStyle"].Index = 3;
			dataSource["ActionsLayout"].Index = 4;
			dataSource["DismissMode"].Index = 5;
			dataSource["DismissDirection"].Index = 6;
			dataSource["AnimationDuration"].Index = 7;
			dataSource["BackgroundDim"].Index = 8;

			dataSource["ActionsLayout"].ValuesProvider = NSArray.FromObjects(new string[] { "Horizontal", "Vertical" });
			dataSource["BackgroundStyle"].ValuesProvider = NSArray.FromObjects(new string[] { "Blur", "Dim" });
			dataSource["DismissMode"].ValuesProvider = NSArray.FromObjects (new string[] { "None", "Tap", "Swipe" });
			dataSource["DismissDirection"].ValuesProvider = NSArray.FromObjects (new string[] { "Horizontal", "Vertical" });

			dataForm = new TKDataForm (this.View.Bounds);
			dataForm.WeakDataSource = this.dataSource.NativeObject;
			dataForm.CommitMode = TKDataFormCommitMode.Manual;
			dataForm.AutoresizingMask = UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleWidth;
			this.View.AddSubview (dataForm);

			if (UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Pad) {
				dataForm.Frame = this.View.Bounds;
				this.View.BackgroundColor = new UIColor(0.937f, 0.937f, 0.957f, 1.00f);
			}

			this.View.BackgroundColor = new UIColor(0.937f, 0.937f, 0.957f, 1.00f);
		}

		public void ShowAlert()
		{
			dataForm.Commit ();

			TKAlert alert = new TKAlert ();

			alert.Title = settings.Title;
			alert.Message = settings.Message;
			alert.AllowParallaxEffect = settings.AllowParallaxEffect;
			alert.Style.BackgroundStyle = settings.BackgroundStyle;
			alert.ActionsLayout = settings.ActionsLayout;
			alert.DismissMode = settings.DismissMode;
			alert.SwipeDismissDirection = settings.DismissDirection;
			alert.AnimationDuration = (float)settings.AnimationDuration;
			alert.Style.BackgroundDimAlpha = (float)settings.BackgroundDim;

			alert.AddActionWithTitle("Shake",  (TKAlert al, TKAlertAction action) => {
				alert.Shake();
				return false;
			});

			alert.AddActionWithTitle("Touch", (TKAlert al, TKAlertAction action) => false);

			alert.AddActionWithTitle("Close",  (TKAlert al, TKAlertAction action) => {
				Console.WriteLine("Close");
				return true;
			});

			alert.Show (true);
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/AlertSettings.swift
//
//  AlertViewSettings.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

class AlertSettings: TKExamplesExampleViewController, TKAlertDelegate, TKDataFormDelegate {
    
    let settings: AlertSettingsInfo = AlertSettingsInfo()
    let dataForm: TKDataForm = TKDataForm()
    let dataSource: TKDataFormEntityDataSource = TKDataFormEntityDataSource()

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Show Alert", action: show)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        
        super.viewDidLoad()

        if UIDevice.currentDevice().userInterfaceIdiom == UIUserInterfaceIdiom.Pad {
            dataForm.frame = self.view.bounds
            self.view.backgroundColor = UIColor(red:0.937, green:0.937, blue:0.957, alpha:1.00)
        }
        else {
            dataForm.frame = self.view.bounds
        }
        
        dataSource.sourceObject = self.settings
        
        dataSource["dismissDirection"].editorClass = TKDataFormSegmentedEditor.self
        dataSource["dismissDirection"].valuesProvider = ["Horizontal", "Vertical"]
        dataSource["dismissMode"].editorClass = TKDataFormSegmentedEditor.self
        dataSource["dismissMode"].valuesProvider = ["None", "Tap", "Swipe"]
        dataSource["actionsLayout"].editorClass = TKDataFormSegmentedEditor.self
        dataSource["actionsLayout"].valuesProvider =  ["Horizontal", "Vertical"]
        dataSource["backgroundStyle"].editorClass = TKDataFormSegmentedEditor.self
        dataSource["backgroundStyle"].valuesProvider = ["Blur", "Dim"]
        
        dataForm.delegate = self
        dataForm.commitMode = TKDataFormCommitMode.Manual
        dataForm.autoresizingMask = UIViewAutoresizing.FlexibleWidth
        self.view.addSubview(dataForm)
        

        dataForm.dataSource = dataSource;
    }
    
    func show()
    {
        dataForm.commit()
        let alert : TKAlert = TKAlert()
        alert.title = settings.title
        alert.message = settings.message
        alert.allowParallaxEffect = settings.allowParallax
        alert.style.backgroundStyle = settings.backgroundStyle
        alert.actionsLayout = settings.actionsLayout
        alert.dismissMode = settings.dismissMode
        alert.swipeDismissDirection = settings.dismissDirection
        alert.animationDuration = settings.animationDuration
        alert.style.backgroundDimAlpha = settings.backgroundDim

        alert.addActionWithTitle("Shake") { (TKAlert, TKAlertAction) -> Bool in
            alert.shake()
            return false
        }
        
        alert.addActionWithTitle("Touch") { (TKAlert, TKAlertAction) -> Bool in
            
            return false
        }
        
        alert.addActionWithTitle("Close") { (TKAlert, TKAlertAction) -> Bool in
            NSLog("closed")
            return true
        }
        
        alert.show(true);
    }
}
